fastapi/routing.py

```diff
@@ -253,17 +253,16 @@ def __init__(
         name: str = None,
         dependency_overrides_provider: Any = None,
     ) -> None:
-        self.path = path
-        self.endpoint = endpoint
-        self.name = get_name(endpoint) if name is None else name
+        super().__init__(
+            path=path, endpoint=endpoint, name=name,
+        )
```

Comment:
Agree, with you. 
But it will be called only at the start-up time so it should be a big performance issue.
And we remove code duplication (init logic from base class).

@SirTelemak what is your opinion?

Reply:
I agree with you about removing duplication and that's great catch and it not making performance issues at all currently since it just return coroutine (not the result). BUT! Starlette is external module and logic can be changed in any moment and that's makes me nervous a bit :) @Kludex what do you think about it?

Reply:
> Starlette is external module and logic can be changed in any moment and that's makes me nervous a bit :) 
We set the package versions, so I don't see that as an issue. If Starlette changes, we need to adapt to keep following the latest Starlette :stuck_out_tongue:

Reply:
Like Pydantic version when update broke python 3.6 CI pipeline? 🤣